pkg/cmd/pr/status/http.go

```diff
@@ -47,17 +49,30 @@ func pullRequestStatus(httpClient *http.Client, repo ghrepo.Interface, options r
 		ReviewRequested edges
 	}
 
+	// SPIKE: Let's see if we can detect here and wire through
+	cachedClient := api.NewCachedHTTPClient(httpClient, time.Hour*24)
+	detector := fd.NewDetector(cachedClient, repo.RepoHost())
+	prFeatures, err := detector.PullRequestFeatures()
```

Comment:
One kind of not great thing about this implementation is that we never use the `MergeQueue` feature as part of this command, and although we can elide the request for this new feature when using GitHub.com (because it's guaranteed to have the new fields), we can't do it for `MergeQueue`. Therefore, we're making unnecessary requests when running this command and pointing at GitHub.com.

Perhaps we should split `PullRequestFeatures` into two different calls?

Reply:
If we combine the feature detection into one request as I mentioned in an above comment we don't need to worry about firing off multiple requests.

Reply:
We should also investigate if we can add back in the `MergeQueue` short circuit on dotcom.

Reply:
> If we combine the feature detection into one request as I mentioned in an above comment we don't need to worry about firing off multiple requests.

I'm not sure this addressed my point. The fact is that to support `gh pr status` on GitHub.com we shouldn't need to make _any_ introspection calls. We don't care about the merge queue status and we know the new summary counts are supported. However, because `PullRequestFeatures` doesn't know that it's `gh pr status` asking rather than something that does need `MergeQueue`, we make the call. My suggestion is that perhaps `PullRequestFeatures` is too broad a collection and should be split.

> We should also investigate if we can add back in the MergeQueue short circuit on dotcom.

If this were possible we could keep the same. I tracked the [mentioned issue in code](https://github.com/cli/cli/issues/5778) and I think the underlying [GQL issue still exists](https://github.com/github/ecosystem-api/issues/3156) but honestly I don't fully have context. Perhaps @mislav can speak to it?

Under the assumption that it is not fixed, what do you think we should do here?

Reply:
Ahh yeah I missed the point a bit. I understand now. I don't have a strong opinion either way, keeping it one function or splitting it up. I would probably lean towards keeping it one function as that is slightly simpler for the caller and would not need any refactoring of places it is currently used. Additionally, the requests are cached for 24 hours so I don't see a huge performance benefit.

Reply:
I'm honestly not sure if the `isInMergeQueue` field is now accessible by PATs rather than only via our OAuth app. Maybe this has been fixed but the platform issue accidentally remained open. We could continue introspecting that field, just in case. But I do not feel strongly that PullRequestFeatures should be split into multiple smaller calls.